src/decode.rs

```diff
@@ -3525,280 +3525,231 @@ unsafe fn decode_b(
-            && b.c2rust_unnamed.c2rust_unnamed.y_mode as libc::c_int
-                <= VERT_LEFT_PRED as libc::c_int
+            && b.y_mode() as IntraPredMode >= VERT_PRED
+            && b.y_mode() as IntraPredMode <= VERT_LEFT_PRED
         {
-            let acdf: *mut uint16_t =
-                (ts.cdf.m.angle_delta[(b.c2rust_unnamed.c2rust_unnamed.y_mode as libc::c_int
-                    - VERT_PRED as libc::c_int) as usize])
-                    .as_mut_ptr();
-            let angle =
-                dav1d_msac_decode_symbol_adapt8(&mut ts.msac, acdf, 6 as libc::c_int as size_t)
-                    as libc::c_int;
-            b.c2rust_unnamed.c2rust_unnamed.y_angle = (angle - 3) as int8_t;
+            let acdf = &mut ts.cdf.m.angle_delta[b.y_mode() as usize - VERT_PRED as usize];
+            let angle = dav1d_msac_decode_symbol_adapt8(&mut ts.msac, acdf.as_mut_ptr(), 6);
+            *b.y_angle_mut() = angle as int8_t - 3;
         } else {
-            b.c2rust_unnamed.c2rust_unnamed.y_angle = 0 as libc::c_int as int8_t;
+            *b.y_angle_mut() = 0;
         }
+
         if has_chroma {
             let cfl_allowed = if frame_hdr.segmentation.lossless[b.seg_id as usize] != 0 {
-                (cbw4 == 1 && cbh4 == 1) as libc::c_int
+                cbw4 == 1 && cbh4 == 1
             } else {
-                (cfl_allowed_mask & ((1 as libc::c_int) << bs as libc::c_uint) as libc::c_uint != 0)
-                    as libc::c_int
+                cfl_allowed_mask & (1 << bs) != 0
             };
-            let uvmode_cdf: *mut uint16_t = (ts.cdf.m.uv_mode[cfl_allowed as usize]
-                [b.c2rust_unnamed.c2rust_unnamed.y_mode as usize])
-                .as_mut_ptr();
+
+            let uvmode_cdf = &mut ts.cdf.m.uv_mode[cfl_allowed as usize][b.y_mode() as usize];
             b.c2rust_unnamed.c2rust_unnamed.uv_mode = dav1d_msac_decode_symbol_adapt16(
                 &mut ts.msac,
-                uvmode_cdf,
-                (N_UV_INTRA_PRED_MODES as libc::c_int - 1 - (cfl_allowed == 0) as libc::c_int)
-                    as size_t,
+                uvmode_cdf.as_mut_ptr(),
+                N_UV_INTRA_PRED_MODES as size_t - 1 - (!cfl_allowed) as size_t,
             ) as uint8_t;
+
             if DEBUG_BLOCK_INFO(f, t) {
-                printf(
-                    b"Post-uvmode[%d]: r=%d\n\0" as *const u8 as *const libc::c_char,
-                    b.c2rust_unnamed.c2rust_unnamed.uv_mode as libc::c_int,
-                    ts.msac.rng,
-                );
+                println!("Post-uvmode[{}]: r={}", b.uv_mode(), ts.msac.rng);
             }
-            b.c2rust_unnamed.c2rust_unnamed.uv_angle = 0 as libc::c_int as int8_t;
-            if b.c2rust_unnamed.c2rust_unnamed.uv_mode as libc::c_int == CFL_PRED as libc::c_int {
-                let sign = (dav1d_msac_decode_symbol_adapt8(
+
+            *b.uv_angle_mut() = 0;
+            if b.uv_mode() as u32 == CFL_PRED {
```

Comment:
Should this be `as` whatever enum `CFL_PRED` is part of?

Reply:
I decided to go the other direction and cast `CFL_PRED` to `u8`, since I think that's the pattern I followed in most other places.